source/MaterialXView/Viewer.cpp

```diff
@@ -1412,7 +1412,11 @@ void Viewer::saveShaderSource(mx::GenContext& context)
         mx::TypedElementPtr elem = material ? material->getElement() : nullptr;
         if (elem)
         {
-            mx::ShaderPtr shader = createShader(elem->getNamePath(), context, elem);
+            const bool hasTransparency = mx::isTransparentSurface(elem, context.getShaderGenerator().getTarget());
+            mx::GenContext materialContext = context;
+            materialContext.getOptions().hwTransparency = hasTransparency;
```

Comment:
Leaving note that this is "fragile" and is value dependent on what isTransparentSurface() returns. 
- Currently opacity / transmissions are detected but transmission_color is not -- should this be fixed as part of this  PR ?
- Also recompiles only occur on load / save shader but not for value changes which can
change the result of this. Defer this to a new issue on how to deal with transparency
update and codegen. (#859)

Reply:
Yes, as discussed this should be looked at in another change list, so thanks for adding a separate issue for that.

I don't think it's worth updating the general `isTransparentSurface()` function to detect more specific inputs at this point. It may be better to wait for the larger transparency overhaul. Ignoring the input `transmission_color` makes it less conservative since it's enough to have `transmission > 0` to detect that transmission handling is needed. So there may be false positives but at least we avoid false negatives.

Reply:
Yes, `transmission_color` is secondary to `transmission` so agree let's defer this. Thanks.